nixos/tests/clevis.nix

```diff
@@ -0,0 +1,440 @@
+{ system ? builtins.currentSystem
+, config ? { }
+, pkgs ? import ../.. { inherit system config; }
+}:
+
+with import ../lib/testing-python.nix { inherit system pkgs; };
+with pkgs.lib;
+
+# The test script boots a NixOS VM, installs NixOS on an empty hard
+# disk, and then reboot from the hard disk. It's parameterized with
+# a test script fragment `createRootPartition', which must create
+# partitions and filesystems.
+
+let
+  makeClevisTest = { name, createRootPartition, extraInstallerConfig ? { }, extraConfig ? "", systemdStage1 ? false, fallback ? false, preBootCommands ? "" }: makeTest {
+    name = "clevis-${name}${optionalString systemdStage1 "-systemd"}${optionalString fallback "-fallback"}";
+    meta.maintainers = with maintainers; [ camillemndn julienmalka ];
+    enableOCR = fallback;
+
+    nodes = {
+      # The configuration of the Tang server.
+      tang = _: {
+        services.tang = {
+          enable = true;
+          listenStream = [ "80" ];
+          ipAddressAllow = [ "192.168.1.0/24" ];
+        };
+        networking.firewall.allowedTCPPorts = [ 80 ];
+      };
+
+      # The configuration of the installer.
+      installer = { pkgs, ... }: {
+        imports = [
+          ../modules/profiles/installation-device.nix
+          ../modules/profiles/base.nix
+          ./common/auto-format-root-device.nix
+          extraInstallerConfig
+        ];
+
+        # The installer is going to create the clevis secret.
+        environment.systemPackages = [ pkgs.clevis ];
+
+        boot.initrd = {
+          availableKernelModules = [ "tpm_tis" ];
+          clevis = { enable = true; useTang = true; };
+          network.enable = !systemdStage1;
+          systemd = {
+            enable = systemdStage1;
+            network.enable = systemdStage1;
+            network.networks."10-static" = {
+              matchConfig.Name = [ "en*" "eth*" ];
+              DHCP = "no";
+              address = [ "192.168.1.1/24" ];
+              linkConfig.RequiredForOnline = true;
+              networkConfig.IPv6PrivacyExtensions = "kernel";
+            };
+          };
+        };
```

Comment:
why does the *installer* need any of this?

Reply:
It does not, but having the installer with a configuration very close to the machine it is installing is helping a lot to reduce the amount of dependencies to list into `system.extraDependencies`, which are actually very painful to find.